examples/constrained/binary_kernel_svm_with_intercept.py

```diff
@@ -19,72 +19,193 @@
 The dual objective of binary kernel SVMs with an intercept contains both
 box constraints and an equality constraint, making it challenging to solve.
 The state-of-the-art algorithm to solve this objective is SMO (Sequential
-minimal optimization). We nevertheless demonstrate in this example how to solve
+minimal optimization). 
+
+We nevertheless demonstrate in this example how to solve
 it by projected gradient descent, by projecting on the constraint set
-using projection_box_section.
+using projection_box_section.  
+  
+Since the dual objective is a Quadratic Program we show how to solve
+it with BoxOSQP too.
 """
 
 from absl import app
+from absl import flags
+
+import jax
 import jax.numpy as jnp
 from jaxopt import projection
 from jaxopt import ProjectedGradient
+from jaxopt import BoxOSQP
+
 import numpy as onp
 from sklearn import datasets
 from sklearn import preprocessing
 from sklearn import svm
 
 
-def objective_fun(beta, lam, K, y):
-  """Dual objective of binary kernel SVMs with intercept."""
+flags.DEFINE_float("lam", 0.5, "Regularization parameter. Must be positive.")
+flags.DEFINE_float("tol", 1e-6, "Tolerance of solvers.")
+flags.DEFINE_integer("num_samples", 30, "Size of train set.")
+flags.DEFINE_integer("num_features", 5, "Features dimension.")
+flags.DEFINE_bool("verbose", False, "Verbosity.")
+FLAGS = flags.FLAGS
+
+
+class LinearKernel:
```

Comment:
I would prefer if we don't introduce classes for this simple case. This obfuscates the code IMO...

Reply:
I agree; let's remove them for now then.